tests/suites/test_suite_pk.function

```diff
@@ -69,6 +69,322 @@ size_t mbedtls_rsa_key_len_func( void *ctx )
  * END_DEPENDENCIES
  */
 
+/* BEGIN_CASE */
+void valid_parameters( )
+{
+    mbedtls_pk_context pk;
+    size_t len;
+
+    mbedtls_pk_init( &pk );
+
+    TEST_VALID_PARAM( mbedtls_pk_setup( &pk, NULL ) ==
+                      MBEDTLS_ERR_PK_BAD_INPUT_DATA );
+
+    TEST_VALID_PARAM( mbedtls_pk_get_bitlen( NULL ) == 0 );
+
+    TEST_VALID_PARAM( mbedtls_pk_get_len( NULL ) == 0 );
+
+    TEST_VALID_PARAM( mbedtls_pk_can_do( NULL, MBEDTLS_PK_NONE ) == 0 );
```

Comment:
As mentioned, I think it should be forbidden to pass `NULL` to `mbedtls_pk_can_do()`, as for the rest of the PK API. In particular, this test should be moved to `invalid_parameters()` as a negative test.

Reply:
I'm keeping the test: this is undocumented behavior, but behavior that we shouldn't change without meaning to. I'll add a comment to this effect.